mereswine/crawler.py

```diff
@@ -44,9 +32,31 @@ def crawl(instance):
 
 
 def geolocate(instance):
-    url = 'http://freegeoip.net/json/{0}'.format(trim_url(instance.url))
-    r = requests.get(url)
+    api_key = current_app.config.get('IP_GEOLOCATION_API_KEY')
+
+    if not api_key:
+        logging.warn('Geolocation API key not set. Geolocation is off.')
+        return
+
+    url = urlparse(instance.url).hostname
+
+    try:
+        ip_address = socket.gethostbyname(url)
+    except socket.gaierror:
+        logging.warn("Couldn't geolocate {}".format(url))
```

Comment:
`logging.warn("Couldn't geolocate %s", url)` (only relevant if we ever use sentry etc)

Reply:
Indeed. Thanks!